File: Examples/GroupDocs.Metadata.Examples.VBasic/Utilities/Tempclass.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports GroupDocs.Metadata.Formats.Document
Imports GroupDocs.Metadata.Standards.Doc 
Imports GroupDocs.Metadata.MetadataProperties
Imports GroupDocs.Metadata.Exceptions

Namespace Utilities
    Class Tempclass
        Public Shared Sub GetDocumentProperties(filePath As String)
            'ExStart:InvalidFormatException
            Try
                ' initialize DocFormat
                Dim docFormat As New DocFormat(Common.MapSourceFilePath(filePath))

                ' initialize metadata
                Dim docMetadata As DocMetadata = docFormat.DocumentProperties

                ' get properties
                Console.WriteLine("Built-in Properties: ")
                For Each [property] As KeyValuePair(Of String, PropertyValue) In docMetadata
                    ' check if built-in property
                    If docMetadata.IsBuiltIn([property].Key) Then
                        Console.WriteLine("{0} : {1}", [property].Key, [property].Value)
                    End If
                Next
            Catch ex As InvalidFormatException
                Console.WriteLine("File is not DOC: {0}", ex.Message)
            End Try
            'ExEnd:InvalidFormatException
        End Sub
    End Class
End Namespace
 

File: Examples/GroupDocs.Metadata.Examples.CSharp/Utilities/Tempclass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using GroupDocs.Metadata.Formats.Document;
using GroupDocs.Metadata.Standards.Doc;
using GroupDocs.Metadata.Examples.Utilities.CSharp;
using GroupDocs.Metadata.MetadataProperties;
using GroupDocs.Metadata.Exceptions;

namespace GroupDocs.Metadata.Examples.Utilities.CSharp
{
    class Tempclass
    {
        public static void GetDocumentProperties(string filePath)
        {
            //ExStart:InvalidFormatException
            try
            {
                // initialize DocFormat
                DocFormat docFormat = new DocFormat(Common.MapSourceFilePath(filePath));

                // initialize metadata
                DocMetadata docMetadata = docFormat.DocumentProperties;

                // get properties
                Console.WriteLine("Built-in Properties: ");
                foreach (KeyValuePair<string, PropertyValue> property in docMetadata)
                {
                    // check if built-in property
                    if (docMetadata.IsBuiltIn(property.Key))
                    {
                        Console.WriteLine("{0} : {1}", property.Key, property.Value);
                    }
                } 
            }
            catch (InvalidFormatException ex)
            {
                Console.WriteLine("File is not DOC: {0}", ex.Message);
            }
            //ExEnd:InvalidFormatException
        }
    }
}
